lib/jarvis/mixins/pool_delegate.rb

```diff
@@ -0,0 +1,27 @@
+require "jarvis/workpool"
+
+module Jarvis module Mixins module PoolDelegate
+  def pool_delegate(pool_name, &callback)
+    lambda do |request|
+      begin
+        pool = WorkPool.fetch(pool_name)
+        pool.post do
+          # TODO(sissel): Note what command is being executed and at what time.
+          # TODO(sissel): Redirect $stdout and $stderr
+          begin
+            callback.call(request)
+          rescue => e
+            # TODO(sissel): Mark this job as failed
+            request.reply(t("unhandled exception", :exception => e))
+            request.reply(e.backtrace.join("\n"))
```

Comment:
I think I'll add something to this PR.

Sadly Hipchat XMPP doesn't support XHTML-IM with inbound message, would you like a small wrapper for this code to use the hipchat web api to for the replies? This would allow us to use rich formatted message.